docs/visual-testing/integrations/webdriverio.md

```diff
@@ -0,0 +1,154 @@
+
+# WebdriverIO Integration
+
+## Introduction
+
+This guide requires an existing WebdriverIO project.<br />
+You can alternatively take a look to our [example repository](#example).
+
+Sauce Labs Visual provides an integration with [WebdriverIO](https://webdriver.io/) through a service that you can add to any existing WebdriverIO project.
+
+Sauce Labs Visual adds new commands to the WebdriverIO's `browser` object:
+- `browser.check()`: Takes a screenshot and send it to Sauce Labs Visual for comparison.
+- `browser.checkResults()`: Waits for diff calculations to complete and returns a summary of results.
+
+Sample usage:
+```ts
+    expect((await browser.checkResult()).UNAPPROVED).toBe(2);
+```
+See [definitions](../../visual-testing.md#definitions) to learn more about different diff statuses.
+
+## Quickstart
+
+### Step 1: Add Sauce Labs Visual dependency
+
+Install the Sauce Labs Visual service in your current project.
+
+```sh
+npm install --save @saucelabs/wdio-sauce-visual-service
+```
+
+### Step 2: Add SauceVisualService to your WebdriverIO configuration
+
+Add the `SauceVisualService` to your existing configuration (E.g. `wdio.conf.(js|ts)`):
+
+```ts
+...
+export const config: Options.Testrunner = {
+...
+    services: ['sauce', ['@saucelabs/wdio-sauce-visual-service', {}]],
+...
+}
+```
+
+### Step 3: Add visual tests in your project
+
+Add a check to one of your tests:
+
+```ts
+    describe('Login Flow', () => {
+        it('should login with valid credentials', async () => {
+            ...
+            await browser.check('My Login Page')
+            ...
+        });
+    })
+```
+
+### Step 4: Configure your Sauce Labs credentials
+
+Sauce Labs Visual relies on environment variables for authentications.<br />
+Both `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` need to be set prior starting your WebdriverIO job.
+
+Username and Access Key can be retrieved from https://app.saucelabs.com/user-settings.
+
+```sh
+export SAUCE_USERNAME=__YOUR_SAUCE_USER_NAME__
+export SAUCE_ACCESS_KEY=__YOUR_SAUCE_ACCESS_KEY__
+```
+
+### Step 5: Run the test
+
+Everything is now configured, your tests can be run as any other WebdriverIO project.
+
+Builds will appear on Sauce Labs platform as soon as they have been created by the running tests: https://app.saucelabs.com/visual/builds.
+
+## Advanced usage
+
+### Build attributes
+
+When creating the service in WebdriverIO's configuration, extra fields can be set to define the context, thus acting on which baselines new snapshots will be compared to. ([More info on baseline matching](../sauce-visual.md#baseline-matching))
+
+Options:
+- `buildName`: Name of the build
+- `project`: Name of the project
+- `branch`: Name of branch
+
+They need to be set through the `options` parameter.
+
+Example:
+```ts
+    services: ['sauce', ['@saucelabs/wdio-sauce-visual-service', {
+        buildName: 'Sauce Demo Test',
+        branch: 'main',
+        project: 'WDIO examples'
+    }]],
+```
+
+### Ignored regions
```

Comment:
Please always start with the component based ignore regions (that's the best practice) and the the coordinate

Reply:
done. They should be in proper order now.